plugins/MidiExport/MidiExport.cpp

```diff
@@ -113,6 +114,7 @@ bool MidiExport::tryExport(const TrackContainer::TrackList &tracks,
 			int base_pitch = 0;
 			double base_volume = 1.0;
 			int base_time = 0;
+			int base_beatLen = 16;
```

Comment:
Looks like there are a few places where the default step note length of 16 is hard-coded in. You should probably create a constexpr constant for it and use that instead.

Reply:
I just followed the example of the other variables, that value is more like a default in case there's no stored value on the XML. Should I change it to be `constexpr int base_beatLen = 16;` here and in the other method?

Reply:
No, I meant if you want the default step note length to be a 1/16th note, you shouldn't have "16" hard-coded in 3 or 4 different places. You should probably use a static constexpr member in the `Note` class called `DefaultStepNoteLength` or something like that:

```c++
// Inside Note class in Note.h
static constexpr int DefaultStepNoteLength = 16; // 16th note
```

Then replace every "16" which you're using to represent the default length of step notes with `Note::DefaultStepNoteLength`.

The point of this is to avoid magic numbers and have the default step note length defined in a single place so it's easy to change later if we ever decide to do that.

Reply:
Oh, got it! I wrote the changes here, I'm just double checking because I think `base_beatLen` is actually not related to the default step note length of 1/16th (the one that shows on the piano roll):

> Updates MidiExport to use Note Types
> 
> 	- Now MidiExport is updated to use note types instead of relying
> on negative length notes.
> 	- For that change it was necessary to find a way of letting
> MidiExport know how long step notes should be. **The solution found was to
> add an attribute to the Instrument XML called "beatlen", which would
> hold the number of frames of the instrument's beat. That would be
> converted to ticks, so we could calculate how long the MIDI notes would
> have to be to play the whole step note. If the attribute was not found,
> the default value of 16 ticks would be used as a length of step notes,
> as a fallback.**

I think I'd need another constexpr for the beatLen fallback value. Should I have it on "InstrumentTrack.h"?